velox/common/caching/SimpleLRUCache.h

```diff
@@ -18,271 +18,145 @@
         curSize,
-        pinnedSize,
-        numElements,
         numHits,
         numLookups);
   }
+
+  bool operator==(const SimpleLRUCacheStats& rhs) const {
+    return this->curSize == rhs.curSize && this->maxSize == rhs.maxSize &&
+        this->numHits == rhs.numHits && this->numLookups == rhs.numLookups;
+  }
 };
 
-/// A simple LRU cache that allows each element to occupy an arbitrary
-/// amount of space in the cache. Useful when your the size of the
-/// cached elements can vary a lot; if they are all roughly the same
-/// size something that only tracks the number of elements in the
-/// cache like common/datastruct/LRUCacheMap.h may be better.
+/// A simple wrapper on top of the folly::EvictingCacheMap that tracks
+/// hit/miss counters. Key/Value evicted are immediately destructed.
+/// So the Key/Value should be a value type or self managing lifecycle
+/// shared_ptr.
 ///
 /// NOTE:
 /// 1. NOT Thread-Safe: All the public calls modify internal structures
 /// and hence require external write locks if used from multiple threads.
-/// 2. 'Key' is required to be copyable and movable.
-template <
-    typename Key,
-    typename Value,
-    typename Comparator = std::equal_to<Key>,
-    typename Hash = std::hash<Key>>
+template <typename Key, typename Value>
 class SimpleLRUCache {
  public:
-  /// Constructs a cache of the specified size. This size can represent whatever
-  /// you want -- slots, or bytes, or etc; you provide the size of each element
-  /// whenever you add a new value to the cache. Note that in certain
-  /// circumstances this max_size may be exceeded -- see add().
-  explicit SimpleLRUCache(size_t maxSize);
-
-  /// Frees all owned data. Check-fails if any element remains pinned.
-  ~SimpleLRUCache();
+  /// Constructs a cache of the specified size. The size represents the number
+  /// of entries in the cache.
+  explicit SimpleLRUCache(size_t maxSize, size_t clearSize = 1);
 
-  /// Add a key-value pair that will occupy the provided size, evicting
-  /// older elements repeatedly until enough room is avialable in the cache.
-  /// Returns whether insertion succeeded. If it did, the cache takes
-  /// ownership of |value|. Insertion will fail in two cases:
-  ///   1) There isn't enough room in the cache even after all unpinned
-  ///      elements are freed.
-  ///   2) The key you are adding is already present in the cache. In
-  ///      this case the element currently existing in the cache remains
-  ///      totally unchanged.
-  ///
-  /// If you use size to represent in-memory size, keep in mind that the
-  /// total space used per entry is roughly 2 * key_size + value_size + 30 bytes
-  /// (nonexact because we use a hash map internally, so the ratio of reserved
-  /// slot to used slots will vary).
-  bool add(Key key, Value* value, size_t size);
+  /// Add an item to the cache. Returns true if the item is successfully
+  /// added, false otherwise.
+  bool add(const Key& key, Value& value);
```

Comment:
Why value input is a reference? Shall we pass by const reference as EvictingCacheMap anyway take a copy in its insert call? Thanks!